### PR TITLE
[Serialization] Remark on transitive dependencies and loading requirement

### PR DESCRIPTION
**Pull request by xymus**
Add a remark to `-Rmodule-loading` mode about transitive dependencies before trying to load them, the previous remarks were all after loading the module. The remark shows if a dependency is required, optional, or ignored. 

Here's what the output looks like for the standard libraries where we see mostly required imports and a few implementation-only dependencies:
```
<unknown>:0: remark: transitive dependency on 'SwiftShims' for 'Swift' is required
<unknown>:0: remark: loaded module 'SwiftShims'; source: '.../lib/swift/shims/module.modulemap', loaded: '.../cache/2NSQLI4GBI520/SwiftShims-EESXXQTXCOAV.pcm'
<unknown>:0: remark: loaded module 'Swift'; source: '.../lib/swift/macosx/Swift.swiftmodule/arm64-apple-macos.private.swiftinterface', loaded: '.../Swift.swiftmodule/arm64-apple-macos.swiftmodule'
<unknown>:0: remark: transitive dependency on 'Swift' for '_StringProcessing' is required
<unknown>:0: remark: transitive dependency on '_RegexParser' for '_StringProcessing' is ignored
<unknown>:0: remark: loaded module '_StringProcessing'; source: '.../_StringProcessing.swiftmodule/arm64-apple-macos.private.swiftinterface', loaded: '.../_StringProcessing.swiftmodule/arm64-apple-macos.swiftmodule'
<unknown>:0: remark: loaded module '_SwiftConcurrencyShims'; source: '.../lib/swift/shims/module.modulemap', loaded: '.../cache/2NSQLI4GBI520/_SwiftConcurrencyShims-EESXXQTXCOAV.pcm'
<unknown>:0: remark: transitive dependency on 'Swift' for '_Concurrency' is required
<unknown>:0: remark: transitive dependency on 'SwiftShims' for '_Concurrency' is required
<unknown>:0: remark: transitive dependency on '_SwiftConcurrencyShims' for '_Concurrency' is ignored
<unknown>:0: remark: loaded module '_Concurrency'; source: '.../_Concurrency.swiftmodule/arm64-apple-macos.private.swiftinterface', loaded: '.../_Concurrency.swiftmodule/arm64-apple-macos.swiftmodule'
```

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
@swift-ci Please smoke test macOS

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
Moved the diagnostic to the actual import location when possible.

@swift-ci Please smoke test

File: test/ModuleInterface/loading-remarks.swift [hunk 1]
```diff
@@ -49,8 +49,10 @@ import SwiftDependency
 import SwiftNonResilientDependency
 import DirectMixedDependency
 
+// CHECK: remark: transitive dependency on 'SwiftShims' for 'Swift' is required
```
Comment (bnbarham) [inline test/ModuleInterface/loading-remarks.swift:52]: Thoughts on:
```
'Swift' has <type> transitive dependency on 'SwiftShims'
```
? The current spelling seems a little confusing as to which is the thing depending on what (to me anyway :P).

**Review by bnbarham — APPROVED** [review]

File: test/ModuleInterface/loading-remarks.swift (hunk 1 above)
Comment (xymus) [reply test/ModuleInterface/loading-remarks.swift:52]: Good suggestion, I'll use that wording instead.

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Pull request merged**